Consul/Agent.cs

```diff
@@ -145,6 +145,8 @@ public class AgentService
         public string[] Tags { get; set; }
         public int Port { get; set; }
         public string Address { get; set; }
+
+        public IDictionary<string, ServiceTaggedAddress> TaggedAddresses { get; set; }
```

Comment:
I know that it is technically a dictionary, but on other hand only two keys (`lan` and `wan`) can be used. I wonder if it wouldn't be better if it is defined as a struct so there is no doubt what can and what cannot be a key?:
```
public TaggedAddresses TaggedAddresses { get; set; }
```

```
    public class TaggedAddress
    {
        public string Address { get; set; }
        public int Port { get; set; }
    }

    public class TaggedAddresses
    {
        public TaggedAddress Lan { get; set; }
        public TaggedAddress Wan { get; set; }
    }
 ```
 
 Initialisation of such struct is a little bit easier:
 ```
 var taggedAddresses = new ServiceTaggedAddresses
 {
      Lan = new ServiceTaggedAddress {Address = "127.0.0.1", Port = 80},
      Wan = new ServiceTaggedAddress {Address = "192.168.10.10", Port = 8000},
 }
 ```

Reply:
Yes, WAN and LAN are specified as keys but it is not the restriction, we can use any keys. For example, we can use 'Internal', 'external' keys. Moreover, in a service registration process via a configuration file, there are no restrictions on keys.

the example below works amazing:

```
          var svcID = KVTest.GenerateTestKeyName();
          var registration = new AgentServiceRegistration
            {
                Name = svcID,
                Port = 8000,
                TaggedAddresses = new Dictionary<string, ServiceTaggedAddress>
                {
                    {"lan", new ServiceTaggedAddress {Address = "127.0.0.1", Port = 80}},
                    {"wan", new ServiceTaggedAddress {Address = "192.168.10.10", Port = 8000}},
                    {"internal", new ServiceTaggedAddress {Address = "127.0.0.2", Port = 82}},
                    {"external", new ServiceTaggedAddress {Address = "192.168.10.12", Port = 8002}}
                }
            };

            await _client.Agent.ServiceRegister(registration);
```

Reply:
All right, so let's keep it as dictionary then 👍